app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,33 @@ def show
     puts params
     @post = Post.find(params[:id])
   end
+
+  # form actions
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.post.new
+    render :new, locals: { post: @post }
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.post.new(post_params)
```

Comment:
- [ OPTIONAL ] This works fine, but I would suggest you make use of @current_user to find it